### ISSUE TITLE
Wire.requestFrom() should not be followed by Wire.endTransmission().

### ISSUE THREAD

**Issue by Koepel** [sparkfun/SparkFun_Qwiic_6DoF_LSM6DSO_Arduino_Library#2]
In the file "SparkFunLSM6DSO.cpp", the Wire.requestFrom() is followed by a Wire.endTransmission() in two places. Those two Wire.endTransmission() has to be removed (only those after a Wire.requestFrom).

Explanation: [Common-mistakes#2](https://github.com/Koepel/How-to-use-the-Arduino-Wire-library/wiki/Common-mistakes#2)
See also my [alternative explanation of the functions of the Wire library](https://github.com/Koepel/How-to-use-the-Arduino-Wire-library/wiki/Explanation-of-the-functions-of-the-Wire-library).

A check if reading the data was okay can be done this way:
```
auto n = _i2cPort->requestFrom(I2CAddress, numBytes);
if (n == numBytes) {
  for(size_t i = 0; i < numBytes; i++) {
    outputPointer[i] =  _i2cPort->read(); 
  }
  return IMU_SUCCESS;
} else {
  return IMU_HW_ERROR;
}

```

**Comment by fralbo** [comment]
Good point @Koepel, I removed calls to `_i2cPort->endTransmission()` after `_i2cPort->read()` in `readRegister()` and `readMultipleRegisters()` and I can now read gyro values.

Thanks a lot mate.
